### PR TITLE
NEXUS-4275 - If index download fails due to error then "download index" task should end in "broken" state.

### PR DESCRIPTION
**Pull request by velo**
Basically the changes I made stop eating IOExceptions.

### CODE REVIEW

File: nexus/nexus-core-plugins/nexus-indexer-lucene-plugin/src/main/java/org/sonatype/nexus/index/DefaultIndexerManager.java [hunk 1]
```diff
@@ -1034,6 +1034,14 @@ protected boolean downloadRepositoryIndex( ProxyRepository repository, boolean f
                         RepositoryStringUtils.getFormattedMessage(
                             "Cannot fetch remote index for repository %s, task cancelled.", repository ) );
                 }
+                catch ( IOException e )
+                {
+                    // kept logs since tasks will only log error if debug is enabled
+                    getLogger().warn(
```
Comment (velo) [inline nexus/nexus-core-plugins/nexus-indexer-lucene-plugin/src/main/java/org/sonatype/nexus/index/DefaultIndexerManager.java:1040]: I'm not sure if there is any real value on keeping this logs, I assumed users may have got used to it and didn't wanna change nexus UX.

But, if anyone things that should be removed I agree.

File: nexus/nexus-test-harness/nexus-test-harness-its/src/test/java/org/sonatype/nexus/integrationtests/nexus4854/Nexus4854SiteXmlInfoIT.java [hunk 2]
```diff
@@ -0,0 +1,216 @@
+/**
```
Comment (cstamas) [inline nexus/nexus-test-harness/nexus-test-harness-its/src/test/java/org/sonatype/nexus/integrationtests/nexus4854/Nexus4854SiteXmlInfoIT.java]: What's this file?

Comment (velo) [inline nexus/nexus-test-harness/nexus-test-harness-its/src/test/java/org/sonatype/nexus/integrationtests/nexus4854/Nexus4854SiteXmlInfoIT.java]: I made that while testing NEXUS-4854, but ended to be an external (maven) problem....

**Comment by nabcos** [comment]
+1

**Comment by cstamas** [comment]
:beer:

**Comment by peterlynch** [comment]
next time lets use a mocking library rather than creating a dummy indexer throwing unsupported operation.

**Comment by mcculls** [comment]
Was a feature build run before this was merged in? I see a failure on nexus-oss now: https://builds.sonatype.org/job/nexus-oss/823/ - this is probably due to an incorrect test that depends on the IOException being hidden, but should be fixed asap.

**Pull request merged**